### PR TITLE
Standardises GTDB execution and allow pre-uncompressed GTDB input

### PR DESCRIPTION
**Pull request by jfy133**
To close https://github.com/nf-core/mag/issues/424

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
Need to test, currently GTDB is not being executed at all;

**Comment by jfy133** [comment]
Tests:

- [ ] Standard test profile (i.e., skip GTDBTK ensures no download of database)

     `nextflow run ../main.nf -profile singularity,test --outdir ./results`

- [ ] Standard test profile (still skipping) but with GTDBDTK path still results in no download/execution

     `nextflow run ../main.nf -profile singularity,test --outdir ./results --gtdb_db ~/cache/databases/gtdbtk_r202_data.tar.gz `
     
 - [ ] Run GTDBK with pre-supplied archive tar (should DB prep it, and run gtdbk)
 
     `nextflow run ../main.nf -profile singularity,test --outdir ./results --gtdb_db ~/cache/databases/gtdbtk_r202_data.tar.gz --skip_gtdbk false `

**Comment by jfy133** [comment]
TODO: same for BUSCO

**Comment by jfy133** [comment]
Change my mind, BUSCO is a little more tricky, will do that in a follow up PR

File: nextflow.config [hunk 1]
```diff
@@ -84,7 +84,8 @@ params {
     cat_db_generate                      = false
     cat_official_taxonomy                = false
     save_cat_db                          = false
-    gtdb                                 = "https://data.ace.uq.edu.au/public/gtdb/data/releases/release202/202.0/auxillary_files/gtdbtk_r202_data.tar.gz"
+    skip_gtdbtk                            = false
```
Comment (jfy133) [inline nextflow.config:87]:
```suggestion
    skip_gtdbtk                          = false
```

**Review by prototaxites — COMMENTED** [review]
Changes look fine to me, but don't have time (or a downloaded copy of the GTDB database) to test with!

> Working: but broken database

Can I check what this means?

**Comment by jfy133** [comment]
It was saying something about couldn't find `tigram database' or something like, regardless if I user the auto-downloaded or manual download gtdb release 202 :shrug: unless the error message was a bit funny and it meant it couldn't find features within the data maybe

**Comment by prototaxites** [comment]
Poking around - the r202 database release is listed with a maximum GTDB version compatibility of 1.7.0 - the version in mag dev at the moment is 2.1.1. 

Worth retrying with the R214 or R207v2 release, which are listed as compatible? A little bit of googling suggests Tigram is some kind of HMM database - maybe this has since been added to the GTDB database. Might be we have to bump the default database version as well.

https://ecogenomics.github.io/GTDBTk/installing/index.html

**Comment by jfy133** [comment]
Huh interesting... then will check that later, I guess somehow the module got updated at some point but not te URL?

**Comment by jfy133** [comment]
Thansk for the investgiation :D (will likely finish next week though as teaching all this week)

**Comment by CarsonJM** [comment]
@jfy133 @prototaxites I think it would be great to get the r214 database set as default, since it is a pretty significant increase over r207. I'm happy to work on adding updating that this week if that would be helpful!

**Comment by CarsonJM** [comment]
Thanks for the guidance @jfy133 I will work on that this week and keep you all posted!

**Comment by CarsonJM** [comment]
Finished running all three tests (auto-download, .tar.gz, and directory) all worked great and it looks like all CI tests are going to pass as well. One thought on this would be to add `label "process_high_memory"` to GTDBTK_DB_PREPARATION so that by default it requests a lot of memory. Would that make sense?

**Comment by jfy133** [comment]
Thanks for adding that @CarsonJM ! Why do you think that process needs lots of memory? Isn't it just running `untar`?

**Comment by CarsonJM** [comment]
Good point! @jfy133 I initially ran this without modifying the resource request, and it failed after > 4hrs (our default queues max runtime). When I requested 16 threads and 100GB mem, it completed in 40 min. After your comment I looked at the trace and mem requirement is definitely low! Would it be running faster because more cores were available?

Trace below:
`task_id	hash	native_id	name	status	exit	submit	duration	realtime	%cpu	peak_rss	peak_vmem	rchar	wchar
5	8b/0ce534	62494	NFCORE_MAG:MAG:GTDBTK:GTDBTK_DB_PREPARATION (gtdbtk_r214_data.tar.gz)	COMPLETED	0	2023-08-07 07:33:11.880	40m 25s	40m 23s	37.7%	6.7 MB	11.1 MB	158.6 GB	161.9 GB`

**Comment by jfy133** [comment]
AFAIK it's also single core... so I have no idea. Maybe the first time the node you were sent had lots of RAM intensive jobs going on? 

I didn't have that issue myself personally (40m every time) when running on my laptop, so I'm more inclined to leave it as is for now?

Otherwise, if you're happy with the PR @CarsonJM please give the :heavy_check_mark: and then we can merge, and dare I say it, make the release?

**Review by CarsonJM — APPROVED** [review]

**Pull request merged**